static/js/dropdown_list_widget.js

```diff
@@ -15,153 +18,562 @@ export const DropdownListWidget = function ({
-                if (e.type === "keypress") {
-                    if (e.key === "Enter") {
-                        setting_elem.find(".dropdown-menu").dropdown("toggle");
-                    } else {
-                        return;
-                    }
+DropdownListWidget.prototype.register_event_handlers = function () {
+    $(`#${CSS.escape(this.container_id)} .dropdown-list-body`).on(
+        "click keypress",
+        ".list_item",
+        (e) => {
+            const setting_elem = $(e.currentTarget).closest(
+                `.${CSS.escape(this.widget_name)}_setting`,
+            );
+            if (e.type === "keypress") {
+                if (e.key === "Enter") {
+                    setting_elem.find(".dropdown-menu").dropdown("toggle");
+                } else {
+                    return;
                 }
-                const value = $(this).attr("data-value");
-                update(value);
-            },
-        );
-        $(`#${CSS.escape(container_id)} .dropdown_list_reset_button`).on("click", (e) => {
-            update(null_value);
-            e.preventDefault();
-        });
+            }
+            const value = $(e.currentTarget).attr("data-value");
+            this.update(value);
+        },
+    );
+    $(`#${CSS.escape(this.container_id)} .dropdown_list_reset_button`).on("click", (e) => {
+        this.update(this.null_value);
+        e.preventDefault();
+    });
+};
+
+DropdownListWidget.prototype.setup_dropdown_widget = function (data) {
+    const dropdown_list_body = $(
+        `#${CSS.escape(this.container_id)} .dropdown-list-body`,
+    ).expectOne();
+    const search_input = $(`#${CSS.escape(this.container_id)} .dropdown-search > input[type=text]`);
+    const get_data = () => {
+        if (this.include_current_item) {
+            return data;
+        }
+        return data.filter((x) => x.value !== this.value.toString());
     };
 
-    const setup = () => {
-        // populate the dropdown
+    ListWidget.create(dropdown_list_body, get_data(data), {
+        name: `${CSS.escape(this.widget_name)}_list`,
+        modifier(item) {
+            return render_dropdown_list({item});
+        },
+        filter: {
+            element: search_input,
+            predicate(item, value) {
+                return item.name.toLowerCase().includes(value);
+            },
+        },
+        simplebar_container: $(`#${CSS.escape(this.container_id)} .dropdown-list-wrapper`),
+    });
+};
+
+// Sets the focus to the ListWidget input once the dropdown button is clicked.
+DropdownListWidget.prototype.dropdown_toggle_click_handler = function () {
+    const dropdown_toggle = $(`#${CSS.escape(this.container_id)} .dropdown-toggle`);
+    const search_input = $(`#${CSS.escape(this.container_id)} .dropdown-search > input[type=text]`);
+
+    dropdown_toggle.on("click", () => {
+        search_input.val("").trigger("input");
+    });
+};
+
+DropdownListWidget.prototype.dropdown_focus_events = function () {
```

Comment:
CZO discussion: https://chat.zulip.org/#narrow/stream/6-frontend/topic/Improve.20dropdown.20widget.20focus.

The bigger issue here is that - We need to have a focus event in case of Multiselect dropdown list widget. (As later it will be implemented in recent topic view and users can shift focus from filter button to dropdown item easily).

In that case, we have to remove the custom event binded on search input, and replace it with our custom focus logic.
That's the reason I added the next commit which replaces the deprecated event handlers in bootstrap. (Well, at least for time being till we remove bootstrap!)

Reply:
Dropping that commit as discussed.